core/src/main/java/org/infinispan/interceptors/impl/EntryWrappingInterceptor.java

```diff
@@ -651,7 +696,12 @@ private void checkTopology(InvocationContext ctx, WriteCommand command) {
    /**
     * Locks the value for the keys accessed by the command to avoid being override from a remote get.
     */
-   protected Object setSkipRemoteGetsAndInvokeNextForManyEntriesCommand(InvocationContext ctx, WriteCommand command) {
+   protected Object setSkipRemoteGetsAndInvokeNextForManyEntriesCommand(InvocationContext ctx, WriteCommand command,
+         CompletionStage<Void> delay) {
+      if (CompletionStages.isCompletedSuccessfully(delay)) {
+         return asyncValue(delay).thenApply(ctx, command, (rCtx, rCommand, rValue) ->
```

Comment:
`delay` is already completed; no need to wait :)

Reply:
Another swap of ifs. My copy paste skills were off when doing this :)